test/e2e/workflow/deployment_test.go

```diff
@@ -336,8 +337,71 @@ var _ = Describe("NetworkAddonsConfig", func() {
 			})
 		})
 	})
+	Describe("NMState", func() {
+		var (
+			configSpec cnao.NetworkAddonsConfigSpec
+		)
+		BeforeEach(func() {
+			configSpec = cnao.NetworkAddonsConfigSpec{
+				NMState: &cnao.NMState{},
+			}
+		})
+		JustBeforeEach(func() {
+			// Install nmstate-operator here
+			installNMStateOperator()
+			CheckNMStateOperatorIsReady(5 * time.Minute)
+
+			CreateConfig(gvk, configSpec)
+			CheckConfigCondition(gvk, ConditionAvailable, ConditionTrue, 15*time.Minute, CheckDoNotRepeat)
+		})
+		JustAfterEach(func() {
+			uninstallNMStateOperator()
+		})
+		Context("when it is already deployed", func() {
+			Context("and the nmstate-operator is installed", func() {
+				It("should run nmstate from the operator", func() {
+					By("checking for NMState in nmstate namespace")
+					Eventually(func() error {
+						nmstateHandlerDaemonSet := &v1.DaemonSet{}
+						return framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: NMStateComponent.DaemonSets[0], Namespace: "nmstate"}, nmstateHandlerDaemonSet)
+					}, 5*time.Minute, time.Second).Should(BeNil(), fmt.Sprintf("Timed out waiting for nmstate-operator daemonset"))
+				})
+			})
+		})
+		Context("when it is not already deployed", func() {
+			BeforeEach(func() {
+				configSpec = cnao.NetworkAddonsConfigSpec{}
+			})
+			Context("and the nmstate-operator is installed", func() {
+				It("should run nmstate from the operator", func() {
+					configSpec = cnao.NetworkAddonsConfigSpec{
+						NMState: &cnao.NMState{},
+					}
+					UpdateConfig(gvk, configSpec)
+					CheckConfigCondition(gvk, ConditionAvailable, ConditionTrue, 15*time.Minute, CheckDoNotRepeat)
+					By("checking for NMState in nmstate namespace")
+					Eventually(func() error {
+						nmstateHandlerDaemonSet := &v1.DaemonSet{}
+						return framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: NMStateComponent.DaemonSets[0], Namespace: "nmstate"}, nmstateHandlerDaemonSet)
+					}, 5*time.Minute, time.Second).Should(BeNil(), fmt.Sprintf("Timed out waiting for nmstate-operator daemonset"))
+				})
+			})
+		})
+	})
 })
 
+func installNMStateOperator() {
+	By("Installing kubernetes-nmstate-operator")
+	result, err := kubectl.Kubectl("apply", "-f", "test/manifests/kubernetes-nmstate-operator.yaml")
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Error when running kubectl: %s", result))
+}
+
+func uninstallNMStateOperator() {
+	By("Uninstalling kubernetes-nmstate-operator")
+	result, err := kubectl.Kubectl("delete", "-f", "test/manifests/kubernetes-nmstate-operator.yaml")
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Error when running kubectl: %s", result))
+}
+
```

Comment:
Can we apply the manifest from the kubernetes-nmstate version at components.yaml ?